### ISSUE TITLE
CHECK_63: Error message not displayed correctly

### ISSUE THREAD

**Issue by ConjuringCoffee** [larshp/abapOpenChecks#1112]
I activated [CHECK_63 (ABAP Doc - Check parameter consistency)](https://docs.abapopenchecks.org/checks/63/) but the errors are not displayed properly. The message of each finding is "Unknown error code" and the navigation always leads to the start of the class, not the abap Doc parameter.

Here's an example (the faulty parameter is `param_3`):

```abap
CLASS ycl_aoc_test DEFINITION
  PUBLIC
  FINAL
  CREATE PUBLIC.

  PUBLIC SECTION.
    "! @parameter param_1 |
    "! @parameter param_3 |
    METHODS example
      IMPORTING
        param_1 TYPE i
        param_2 TYPE i.

  PROTECTED SECTION.

  PRIVATE SECTION.
ENDCLASS.


CLASS ycl_aoc_test IMPLEMENTATION.
  METHOD example.
    RETURN.
  ENDMETHOD.
ENDCLASS.
```

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/538bd1db-f594-4132-8c13-d46a7eb5885c)

Any idea why this happens? Thanks!

**Comment by larshp** [comment]
guess something changed in the abapdoc api, try setting a breakpoint

**Comment by ConjuringCoffee** [comment]
The check successfully reaches method `INFORM` with the following text in `p_param_1`: 

`inconsistency ABAP Doc comment - parameter interface of element EXAMPLE`

So it doesn't seem like this is a problem with the abap Doc API to me. Any other ideas?

**Comment by MiLiCLN** [comment]
I suspect the problem comes from the defined error code-number "003" in the constructor and the passed code-number in "inform" (001).
Can you try to make "003" into "001" in the constructor at "insert_scimessage"?
![image](https://github.com/larshp/abapOpenChecks/assets/49416412/f7c9b71c-f675-4aa2-b37d-9975c3a01870)

**Comment by ConjuringCoffee** [comment]
This is the result now:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/73ca0f23-39d4-4cb6-8b39-6abdf835422b)

I think we're on the right path, but we're not quite there yet. Any chance we can exchange the very generic message for a more specific message? The more specific message can only be seen in SAP GUI:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/73763923-559e-4064-95ef-cc33b22e843b)

**Comment by MiLiCLN** [comment]
Ok, step by step a little further. :-)
Unfortunately, I can't test it. But what can be seen in the Datail view?

**Comment by ConjuringCoffee** [comment]
The detail view looks good:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/47412de4-a238-41cd-9ff4-ae8db13bfcba)

Compared to before:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/2a6f9ab5-742f-4d67-8852-914dc61b51e0)

**Comment by MiLiCLN** [comment]
A PR is welcome. :-)

**Comment by ConjuringCoffee** [comment]
I didn't submit a pull request because I'm not happy yet with how the text turns out. Compare it to check 001 for example:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/ada1c3c2-c5b0-4a94-b514-afdb88e3f986)

How can this text be changed? I tried to change `iv_text` like this, but it had no effect:

```abap
    insert_scimessage(
        iv_code = '001'
        iv_text = 'ABAP Doc parameter name is inconsistent' ).
```

**Comment by MiLiCLN** [comment]
What is your goal? To change the highlighted text in the screenshot?
![image](https://github.com/larshp/abapOpenChecks/assets/49416412/2048e3d8-29bf-4f6e-912b-2f4e3e09e3fc)
This text is from the class description; the check-title.

**Comment by ConjuringCoffee** [comment]
I want to change the other text 😊

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/50a2f0e1-e394-4f31-b9cf-a7971a93c0fe)

**Comment by MiLiCLN** [comment]
I had now also the possibility to test this with eclipse. For me the result looks like this (with the latest PR):
![image](https://github.com/larshp/abapOpenChecks/assets/49416412/758ef52a-3a91-4a6f-96dc-86328b2942ca)

**Comment by ConjuringCoffee** [comment]
That's weird! I'll check the installation and report back. (Could take a while, I'm having some authorization problems right now...)

**Comment by ConjuringCoffee** [comment]
My authorization problems are fixed and I'm running the latest version now. The message still isn't displayed correctly. 

Can anyone please re-open this issue?

I also tried it using SAP GUI, same result:
![image](https://github.com/larshp/abapOpenChecks/assets/72548231/5d814b56-598a-465a-971f-bff0baf67663)

**Comment by ConjuringCoffee** [comment]
Are we still talking about the same problem? I'd be actually okay with the way it already works in your previous screenshots, it just doesn't seem to work on my system. 😅

* On your system: `inconsistency ABAP Doc comment - parameter interface of [...]` (cut off in the screenshot)
![image](https://github.com/larshp/abapOpenChecks/assets/72548231/1eb15557-9275-44f0-87cb-6b5c432840a2)
* On my system: `Check: ZCL_AOC_CHECK_63, Rule: 001`
![image](https://github.com/larshp/abapOpenChecks/assets/72548231/70ad908f-cc18-4c21-911b-b9e47902c796)

**Comment by ConjuringCoffee** [comment]
Right, so I have a bit more experience with ATC checks now and I think I know what's going on now.

* The ATC check message is only defined with a message variable: `&1`
* The actual text (`inconsistency ABAP Doc comment - parameter interface [...]` is put into the message variable when calling method `INFORM`
* For some reason, the behavior of [MiLiCLN](https://github.com/MiLiCLN)'s system is different to mine: On his system, ADT already shows the complete message. On my system, it shows `Check: ZCL_AOC_CHECK_63, Rule: 001`
* I conclude that my system replaces the complete message text with `Check: ZCL_AOC_CHECK_63, Rule: 001` if it only consists of a single message variable. If I include any other text in the message, then it works as expected.

I'll provide a pull request to fix this.

**Comment by ConjuringCoffee** [comment]
![image](https://github.com/larshp/abapOpenChecks/assets/72548231/63c827f0-f3cf-4482-bdbf-c1b69423d86e)

I've changed the message from `&1` to `ABAP Doc: &1` and it works in the SAP GUI:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/9a5774ce-217d-4a36-bdce-dd81367b60d0)

However, it doesn't work in ADT:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/0a783deb-617e-4061-b3b6-3e082ac8649f)

If I change the error code from `'001'` to `'002'` (both in the `constructor` and in the `run` method), then it works in ADT too:

![image](https://github.com/larshp/abapOpenChecks/assets/72548231/4d41382b-4f10-4877-9cec-d17f06720fa2)

Restarting my own PC didn't help, and it also behaves the same for colleagues. Maybe there is some kind of additional caching for the ATC message check? I'll keep an eye on it.

**Comment by ConjuringCoffee** [comment]
> Maybe there is some kind of additional caching for the ATC message check? I'll keep an eye on it.

I suspect that this was the case. The error message is correctly displayed for `'001'` now.
